DonutsGo/DonutsGo.Application/Services/ProductService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using DonutsGo.Application.Exceptions;
+using DonutsGo.Application.Models.Products;
+using DonutsGo.DataAccess;
+using DonutsGo.DataAccess.Entities;
+
+namespace DonutsGo.Application.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly DatabaseContext databaseContext;
+
+        public ProductService(DatabaseContext databaseContext)
+        {
+            this.databaseContext =databaseContext;
+        }
+
+        public List<ProductResponseModel>GetAllProducts()
```

Comment:
It looks like the service is not fully used in the ProductsController so I guess this is work in progress and you are trying to replace the in memory storage with database storage.